ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosChangeFeedBasedReplicationFeed.java

```diff
@@ -140,9 +140,11 @@ boolean isExpired() {
    * Constructor to create a {@link CosmosChangeFeedBasedReplicationFeed} object.
    * @param cosmosDataAccessor {@link CosmosDataAccessor} object.
    * @param azureMetrics{@link {@link AzureMetrics} object.
+   * @param changeFeedBatchSize batch size for each change feed request.
    */
-  public CosmosChangeFeedBasedReplicationFeed(CosmosDataAccessor cosmosDataAccessor, AzureMetrics azureMetrics) {
-    this.defaultCacheSize = AzureCloudDestination.getFindSinceQueryLimit();
+  public CosmosChangeFeedBasedReplicationFeed(CosmosDataAccessor cosmosDataAccessor, AzureMetrics azureMetrics,
+      int changeFeedBatchSize) {
+    this.defaultCacheSize = changeFeedBatchSize;
```

Comment:
Why is this called default CacheSize?  Can it be overridden?

Reply:
Its true that it cannot be overridden in the current state of the code. 
I named this variable "default" because in future we might want a feature to tune change feed to fetch more aggressively based on if we are just catching up on the replica tokens, and the blobs are already present.